### DISCUSSION TITLE
One-step interpolation for resampling onto surfaces

### DISCUSSION THREAD

**Discussion by feilong** [nipreps/fmriprep#2813] [Ideas]
Hi fMRIPrep folks,

I wonder if you are interested in a PR that implements one-step interpolation for projecting fMRI data onto surfaces.  My understanding is that currently `mri_vol2surf` takes resampled volumetric data in T1w space as input and resamples it onto surface, which are two steps in total.  One-step interpolation has been implemented for the volume workflow, and it would be nice to have it for the surface workflow, too.

Best,
Feilong

**Comment by effigies** [comment]
I would definitely be interested.

**Reply by feilong** [reply]
Thanks, Chris! I've been trying to implement it in a plugin manner, which reads the working directory of fMRIPrep to get the data and transforms, combines the transforms, and resamples using `scipy.ndimage.map_coordinates`. I'm still testing whether it's working as expected, but in the long run it would be much more elegant to have a built-in workflow like that.

**Reply by effigies** [reply]
@feilong In the long run, I would like to make sure that everything you need from the working directory is in derivatives, so this can be done even without adding it to the workflow (though it would definitely be good). We are starting to work on #2207 which will produce a mode where you only get transforms, and then construct your final BOLD series in a single application step at the end (or after the fact).

**Reply by feilong** [reply]
That would be even more awesome!  It would make things much more easier when we want the same data in a different space, and it doesn't require keeping the working directory or rerunning the workflow.
